src/Files.App/Views/Layouts/BaseLayoutPage.cs

```diff
@@ -903,22 +903,26 @@ private async Task AddShellMenuItemsAsync(List<ContextMenuFlyoutItemViewModel> s
 				}
 			}
 
-			// Add items to main shell submenu
-			mainShellMenuItems.Where(x => x.LoadSubMenuAction is not null).ForEach(async x =>
+			// Filter mainShellMenuItems that have a non-null LoadSubMenuAction
+			var mainItemsWithSubMenu = mainShellMenuItems.Where(x => x.LoadSubMenuAction is not null);
+			
+			var mainSubMenuTasks = mainItemsWithSubMenu.Select(async item =>
 			{
-				await x.LoadSubMenuAction();
-
-				ShellContextFlyoutFactory.AddItemsToMainMenu(mainItems, x);
+				await item.LoadSubMenuAction();
+				ShellContextFlyoutFactory.AddItemsToMainMenu(mainItems, item);
 			});
 
-			// Add items to overflow shell submenu
-			overflowShellMenuItems.Where(x => x.LoadSubMenuAction is not null).ForEach(async x =>
-			{
-				await x.LoadSubMenuAction();
+			// Filter overflowShellMenuItems that have a non-null LoadSubMenuAction
+			var overflowItemsWithSubMenu = overflowShellMenuItems.Where(x => x.LoadSubMenuAction is not null);
 
-				ShellContextFlyoutFactory.AddItemsToOverflowMenu(overflowItem, x);
+			var overflowSubMenuTasks = overflowItemsWithSubMenu.Select(async item =>
+			{
+				await item.LoadSubMenuAction();
+				ShellContextFlyoutFactory.AddItemsToOverflowMenu(overflowItem, item);
 			});
 
+			await Task.WhenAll(mainSubMenuTasks.Concat(overflowSubMenuTasks));
```

Comment:
Sorry, one more thing.
This awaiting should be put in the end of the method, because the following process doesn't need to wait loading submenus.